### PR TITLE
Moved functions from one place to another; added tests

### PR DESCRIPTION
**Pull request by coconutpalm**
Also:
- All the new code here is copy/pasted from BEETL.
- data.clj consists of data conversion functions consolidated from errors.clj and/or patterns.clj.
- Doc in readme about desired features for 1.0

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
             :url "http://www.eclipse.org/legal/epl-v10.html"}
 
-  :plugins [[lein-test-bang-bang "0.2.0"]]
-
-  :profiles {:test {:dependencies [[com.h2database/h2 "1.4.192"]]}
-             :repl {:dependencies [[com.h2database/h2 "1.4.192"]]}}
-
-  :aot :all
+  :plugins [[lein-test-bang-bang "0.2.0"] ; lein test!! - Run each test NS in a separate JRE
+            [lein-ancient "0.6.10"]       ; lein ancient - Check for outdated dependencies
```
Comment (levi-andrew-dixon) [inline project.clj]: I'm not sure I would put this in the JAR unless it is used programatically.  Maybe in the documentation note how to add to your lein dev profile.

File: src/clj_foundation/errors.clj [hunk 2]
```diff
@@ -293,6 +275,8 @@
     value))
 
 
+;; FIXME: The following two do roughly the same thing.  Consolidate.
```
Comment (levi-andrew-dixon) [inline src/clj_foundation/errors.clj:278]: Did you mean to check in the FIXME or fix it before the pull request?

**Comment by levi-andrew-dixon** [comment]
👍

File: src/clj_foundation/errors.clj (hunk 2 above)
Comment (coconutpalm) [reply src/clj_foundation/errors.clj:278]: Good catch.  Thanks

**Comment by greglakomy** [comment]
👍

**Pull request merged**